src/main/java/io/github/thebusybiscuit/slimefun4/core/multiblocks/MultiBlockMachine.java

```diff
@@ -165,6 +171,33 @@ public void load() {
         }
     }
 
+    /**
+     * This method handles an output {@link ItemStack} from the {@link MultiBlockMachine} which has a crafting delay
+     *
+     * @param outputItem
+     *          A crafted {@link ItemStack} from {@link MultiBlockMachine}
+     * @param block
+     *          Main {@link Block} of our {@link Container} from {@link MultiBlockMachine}
+     *
+     */
+    @ParametersAreNonnullByDefault
+    protected void handleCraftedItem(ItemStack outputItem, Block block) {
+        BlockState state = PaperLib.getBlockState(block, false).getState();
+        Validate.isTrue(state instanceof Container);
+        Container container = (Container) state;
+        Inventory containerInv = container.getInventory();
```

Comment:
Well, now we have the same paradox as before, getting the BlockState is an expensive operation and we already gotten the BlockState at this point, so this will hurt performance. The BlockState needs to be passed as an argument as much as the Block does.